packages/editor/src/components/block-settings-menu/with-block-settings-actions.js

```diff
@@ -64,5 +63,10 @@ export default compose(
 				);
 			}
 		},
+		onRemove() {
+			dispatch( 'core/editor' ).removeBlocks( clientIds );
+		},
 	} ) ),
-)( BlockDuplicateButton );
+] );
+
+export default withBlockSettingsActions;
```

Comment:
Can you remind me why we're putting this in its own file, again? Was it just to make `block-settings-menu/index.js` less large?

I'm wondering if it's unnecessary indirection. `withBlockSettingsActions` and `BlockSettingsMenu` are very tightly coupled so it makes sense in my mind that their code remains very near each other.

Reply:
Yeah, totally agree. Initially, I tried out using `withBlockSettingsActions` as a HOC for both the `BlockSettingsMenu` and `BlockSettingsKeyboardShortcuts` (to solve the issue where shortcuts weren't being bound until the menu was rendered). 

I agree now that it's only used to wrap one component they can be combined.

Reply:
Done in 8db7543.

Have also gone one step further and refactored `BlockSettingsKeyboardShortcuts` into `BlockSettingsMenu` in f847517, as it wasn't doing anything.